File: apps/desktop/src-tauri/src/api/sessions.rs
use crate::api::ApiClient;
use crate::error::AppResult;
use crate::models::{CreateSessionDto, SessionDetailResponse, SessionHeader, UpdateSessionDto};

pub async fn list_sessions(client: &ApiClient, cwd: Option<&str>, project_id: Option<&str>, only_deleted: Option<bool>) -> AppResult<Vec<SessionHeader>> {
    let mut query: Vec<(&str, &str)> = Vec::new();
    if let Some(cwd) = cwd {
        query.push(("cwd", cwd));
    }
    if let Some(project_id) = project_id {
        query.push(("projectId", project_id));
    }

    let only_deleted_str = only_deleted.map(|b| if b { "true" } else { "false" });
    if let Some(ref val) = only_deleted_str {
        query.push(("onlyDeleted", val));
    }

    if query.is_empty() {
        client.get("/sessions").await
    } else {
        client.get_with_query("/sessions", &query).await
    }
}

pub async fn create_session(client: &ApiClient, dto: &CreateSessionDto) -> AppResult<SessionHeader> {
    client.post("/sessions", dto).await
}

pub async fn get_session(client: &ApiClient, id: &str) -> AppResult<SessionDetailResponse> {
    client.get(&format!("/sessions/{}", id)).await
}

pub async fn update_session(client: &ApiClient, id: &str, dto: &UpdateSessionDto) -> AppResult<SessionHeader> {
    client.patch(&format!("/sessions/{}", id), dto).await
}

pub async fn delete_session(client: &ApiClient, id: &str) -> AppResult<serde_json::Value> {
    client.delete(&format!("/sessions/{}", id)).await
}

pub async fn restore_session(client: &ApiClient, id: &str) -> AppResult<serde_json::Value> {
    client.post(&format!("/sessions/{}/restore", id), &serde_json::Value::Null).await
}

File: apps/console-macos/Sources/ConsoleCore/Api/Sessions.swift
import Foundation

// Mirrors api/sessions.rs

public extension ApiClient {
    func listSessions(cwd: String? = nil, projectId: String? = nil) async throws -> [SessionHeader] {
        var query: [URLQueryItem] = []
        if let cwd { query.append(.init(name: "cwd", value: cwd)) }
        if let projectId { query.append(.init(name: "projectId", value: projectId)) }
        return try await get("/sessions", query: query)
    }

    func createSession(_ dto: CreateSessionDto) async throws -> SessionHeader {
        try await post("/sessions", body: dto)
    }

    func getSession(id: String) async throws -> SessionDetailResponse {
        try await get("/sessions/\(id)")
    }

    func updateSession(id: String, _ dto: UpdateSessionDto) async throws -> SessionHeader {
        try await patch("/sessions/\(id)", body: dto)
    }

    func deleteSession(id: String) async throws -> JSONValue {
        try await delete("/sessions/\(id)")
    }
}
